### ISSUE TITLE
Allow fit params for LightGBM

### ISSUE THREAD

**Issue by ValeriaGomes** [nubank/fklearn#117]
### Describe the feature and the current state.
- LightGBM has several parameters inside the `fit` method. One possible solution is to add a `fit_params` into the LightGBM function in FkLearn for increased flexibility.

### Will this change a current behavior? How?
We would be able to configure the fit function using FkLearn.

### Additional Information
https://lightgbm.readthedocs.io/en/latest/pythonapi/lightgbm.LGBMClassifier.html#lightgbm.LGBMClassifier.fit

**Comment by sadikneipp** [comment]
I think the extra params argument covers that, right?
[https://github.com/nubank/fklearn/blob/master/src/fklearn/training/classification.py#L507:L511](url)

**Comment by sadikneipp** [comment]
Hmm I'm wrong, this argument only covers the hyperparameter section here https://lightgbm.readthedocs.io/en/latest/Parameters.html

**Comment by sadikneipp** [comment]
Any specific behaviour you feel we are missing @ValeriaGomes? I see there's some overlap between what `extra_params` cover and the params in the link you sent

**Comment by ValeriaGomes** [comment]
Good point!  I hadn't seen the `extra_params`. I'll check if there're parameters missing and if it's not the case, I'll close the issue.

**Comment by robotenique** [comment]
🆙  on this, I think with the current learners we cannot pass a validation set to work with early stopping in LightGBM.
